vendor/bat-native-ledger/src/bat_get_media.cc

```diff
@@ -511,11 +511,39 @@ void BatGetMedia::processYoutubeMediaPanel(uint64_t windowId,
     processYoutubeChannelPath(windowId, visit_data, providerType);
   } else if (visit_data.path.find("/user/") != std::string::npos) {
     processYoutubeUserPath(windowId, visit_data, providerType);
+  } else if (!isYTPath(visit_data.path)) {
+    processYoutubeCustomPath(windowId, visit_data, providerType);
   } else {
     onMediaActivityError(visit_data, providerType, windowId);
   }
 }
 
+bool BatGetMedia::isYTPath(const std::string& path) const {
+  std::vector<const std::string> yt_paths =
+    braveledger_bat_helper::GetYoutubeSpecificPaths();
+
+  // make sure we are ignoring actual YT paths and not
+  // a custom path that might start with a YT path
+  std::string yt_path = getRealEnteredYTPath(path);
+  for (std::string str_path : yt_paths) {
+    if (yt_path == str_path) {
+      return true;
+    }
+  }
+  return false;
+}
+
+std::string BatGetMedia::getRealEnteredYTPath(const std::string& path) const {
```

Comment:
please add unit tests for it

Reply:
Are we adding tests for private members?

Reply:
you can also test `getMediaActivityFromUrl` and cover this private method scenarios

Reply:
Done.